### PR TITLE
TypeObjectRegistry refactor [21132]

### PR DESCRIPTION
**Pull request by richiware**
## Description

With this refactor, any registration in `TypeObjectRegistry` will return a `TypeIdentifierPair` which a `TopicDataType` can use to configure internally DDS.

Next PRs depends on this one:
- eprosima/shapesdemo#139
- eProsima/Fast-DDS-python#132
- eProsima/Fast-DDS-docs#792

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- [x] Any new/modified methods have been properly documented using Doxygen. 
- *N/A* Any new configuration API has an equivalent XML API (with the corresponding XSD extension) 
- *N/A* Changes are backport compatible: they do **NOT** break ABI nor change library core behavior. 
- *N/A* Changes are API compatible. 
- *N/A* New feature has been added to the `versions.md` file (if applicable).
- *N/A* New feature has been documented/Current behavior is correctly described in the documentation. 
    
- *N/A* Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Review by adriancampo — CHANGES_REQUESTED** [review]

File: include/fastdds/dds/topic/TopicDataType.hpp
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/topic/TopicDataType.hpp:1]: Needed [here](https://github.com/eProsima/Fast-DDS/pull/4874/files#diff-c0abde0afeb3603890dfa3ac2dfa8b55d1f73f68bec2bc642cc5b415497a09e1R1652).

File: include/fastdds/dds/xtypes/common.hpp [hunk 1]
```diff
@@ -35,11 +35,10 @@ enum class ExtensibilityKind : uint32_t
     MUTABLE
 };
 
-enum class TryConstructKind : uint32_t
+enum class TryConstructFailAction : uint32_t
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/common.hpp:37]: `TryConstructKind` is defined in Dynamic Language Binding and `TryConstructFailAction` is the enumeration for the annotation used to know the TryConstruct.

File: src/cpp/fastdds/xtypes/dynamic_types/MemberDescriptorImpl.cpp [hunk 2]
```diff
@@ -269,18 +268,15 @@ bool MemberDescriptorImpl::is_consistent() noexcept
...
+            EPROSIMA_LOG_WARNING(DYN_TYPES,
                     "try_construct_kind is set but parent type of member is not TK_STRUCTURE or TK_UNION.");
-            return false;
-        }
-        else
-        {
-            EPROSIMA_LOG_WARNING(DYN_TYPES, "try_construct_kind is not implemented yet.");
         }
+
+        EPROSIMA_LOG_WARNING(DYN_TYPES, "try_construct_kind is not implemented yet.");
```
[the remark this answers is not in the document]
Comment (richiware) [reply src/cpp/fastdds/xtypes/dynamic_types/MemberDescriptorImpl.cpp]: Old comment. I changed the behaviour.

File: include/fastdds/dds/xtypes/type_representation/ITypeObjectRegistry.hpp [hunk 3]
```diff
@@ -54,7 +54,8 @@ class ITypeObjectRegistry
      *
      * @param[in] type_name Name of the type being registered.
      * @param[in] complete_type_object CompleteTypeObject related to the given type name.
-     * @param[out] type_id TypeIdentifier corresponding to the CompleteTypeObject just registered.
+     * @param[out] type_ids TypeIdentifiers corresponding to the CompleteTypeObject just registered and the
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/type_representation/ITypeObjectRegistry.hpp]: Done in eb4efe39b

File: include/fastdds/dds/xtypes/type_representation/ITypeObjectRegistry.hpp [hunk 4]
```diff
@@ -64,18 +65,42 @@ class ITypeObjectRegistry
...
+    virtual ReturnCode_t register_type_object(
+            const TypeObject& type_object,
+            TypeIdentifierPair& type_ids) = 0;
 
     /**
      * @brief Register DynamicType TypeObject.
      *
      * @param[in] dynamic_type DynamicType to be registered.
-     * @param[out] type_id TypeIdentifier corresponding to the registered DynamicType TypeObject.
+     * @param[out] type_ids TypeIdentifiers corresponding to the registered DynamicType TypeObject.
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/type_representation/ITypeObjectRegistry.hpp]: Done in eb4efe39b

File: include/fastdds/dds/xtypes/type_representation/ITypeObjectRegistry.hpp [hunk 5]
```diff
@@ -85,7 +110,8 @@ class ITypeObjectRegistry
      * @pre type_name must not be empty.
      *
      * @param[in] type_name Name of the type being registered.
-     * @param[in] type_identifier TypeIdentier related to the given type name.
+     * @param[inout] type_identifier TypeIdentifier related to the given type name. It must be set in
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/type_representation/ITypeObjectRegistry.hpp]: Done in eb4efe39b

File: include/fastdds/dds/xtypes/type_representation/ITypeObjectRegistry.hpp [hunk 6]
```diff
@@ -144,6 +170,23 @@ class ITypeObjectRegistry
             const TypeIdentifier& type_identifier,
             TypeObject& type_object) = 0;
 
+    /**
+     * @brief Get the TypeInformation related to a specific type_name.
+     *
+     * @pre type_ids must not be empty.
+     *
+     * @param[in] type_ids @ref TypeIdentifier which type information is queried.
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/type_representation/ITypeObjectRegistry.hpp]: Done in eb4efe39b

File: include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp [hunk 7]
```diff
@@ -391,6 +394,7 @@ class TypeObjectUtils
      *
      * @param[in] string StringSTypeDefn union member to set.
      * @param[in] type_name Type name to be registered.
+     * @param[out] type_ids TypeIdentifiers corresponding to the StringSTypeDefn just registered.
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp]: Done in eb4efe39b

File: include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp [hunk 8]
```diff
@@ -402,13 +406,15 @@ class TypeObjectUtils
...
             const std::string& type_name,
+            TypeIdentifierPair& type_ids,
             bool wstring = false);
 
     /**
      * @brief Register large string/wstring TypeIdentifier into TypeObjectRegistry.
      *
      * @param[in] string StringLTypeDefn union member to set.
      * @param[in] type_name Type name to be registered.
+     * @param[out] type_ids TypeIdentifiers corresponding to the StringLTypeDefn just registered.
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp]: Done in eb4efe39b

File: include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp [hunk 9]
```diff
@@ -420,13 +426,15 @@ class TypeObjectUtils
...
             const std::string& type_name,
+            TypeIdentifierPair& type_ids,
             bool wstring = false);
 
     /**
      * @brief Register small sequence TypeIdentifier into TypeObjectRegistry.
      *
      * @param[in] plain_seq PlainSequenceSElemDefn union member to set.
      * @param[in] type_name Type name to be registered.
+     * @param[out] type_ids TypeIdentifiers corresponding to the PlainSequenceSElemDefn just registered.
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp]: Done in eb4efe39b

File: include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp [hunk 10]
```diff
@@ -436,13 +444,15 @@ class TypeObjectUtils
...
-            const std::string& type_name);
+            const std::string& type_name,
+            TypeIdentifierPair& type_ids);
 
     /**
      * @brief Register large sequence TypeIdentifier into TypeObjectRegistry.
      *
      * @param[in] plain_seq PlainSequenceLElemDefn union member to set.
      * @param[in] type_name Type name to be registered.
+     * @param[out] type_ids TypeIdentifiers corresponding to the PlainSequenceLElemDefn just registered.
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp]: Done in eb4efe39b

File: include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp [hunk 11]
```diff
@@ -452,13 +462,15 @@ class TypeObjectUtils
...
-            const std::string& type_name);
+            const std::string& type_name,
+            TypeIdentifierPair& type_ids);
 
     /**
      * @brief Register small array TypeIdentifier into TypeObjectRegistry.
      *
      * @param[in] plain_array PlainArraySElemDefn union member to set.
      * @param[in] type_name Type name to be registered.
+     * @param[out] type_ids TypeIdentifiers corresponding to the PlainArraySElemDefn just registered.
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp]: Done in eb4efe39b

File: include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp [hunk 12]
```diff
@@ -468,13 +480,15 @@ class TypeObjectUtils
...
-            const std::string& type_name);
+            const std::string& type_name,
+            TypeIdentifierPair& type_ids);
 
     /**
      * @brief Register large array TypeIdentifier into TypeObjectRegistry.
      *
      * @param[in] plain_array PlainArrayLElemDefn union member to set.
      * @param[in] type_name Type name to be registered.
+     * @param[out] type_ids TypeIdentifiers corresponding to the PlainArrayLElemDefn just registered.
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp]: Done in eb4efe39b

File: include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp [hunk 13]
```diff
@@ -484,13 +498,15 @@ class TypeObjectUtils
...
-            const std::string& type_name);
+            const std::string& type_name,
+            TypeIdentifierPair& type_ids);
 
     /**
      * @brief Register small map TypeIdentifier into TypeObjectRegistry.
      *
      * @param[in] plain_map PlainMapSTypeDefn union member to set.
      * @param[in] type_name Type name to be registered.
+     * @param[out] type_ids TypeIdentifiers corresponding to the PlainMapSTypeDefn just registered.
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp]: Done in eb4efe39b

File: include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp [hunk 14]
```diff
@@ -500,13 +516,15 @@ class TypeObjectUtils
...
-            const std::string& type_name);
+            const std::string& type_name,
+            TypeIdentifierPair& type_ids);
 
     /**
      * @brief Register large map TypeIdentifier into TypeObjectRegistry.
      *
      * @param[in] plain_map PlainMapLTypeDefn union member to set.
      * @param[in] type_name Type name to be registered.
+     * @param[out] type_ids TypeIdentifiers corresponding to the PlainMapLTypeDefn just registered.
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp]: Done in eb4efe39b

File: include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp [hunk 15]
```diff
@@ -1691,6 +1723,8 @@ class TypeObjectUtils
      *
      * @param[in] alias_type CompleteAliasType.
      * @param[in] type_name Name to be registered in the registry.
+     * @param[out] type_ids TypeIdentifiers corresponding to the CompleteAliasType just registered and the
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp]: Done in eb4efe39b

File: include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp [hunk 16]
```diff
@@ -1700,15 +1734,17 @@ class TypeObjectUtils
...
+            TypeIdentifierPair& type_ids);
 
     /**
      * @brief Register annotation TypeObject into TypeObjectRegistry.
      *        CompleteAnnotationType is provided and the minimal TypeObject is constructed from the complete one.
      *
      * @param[in] annotation_type CompleteAnnotationType.
      * @param[in] type_name Name to be registered in the registry.
-     * @param[out] type_id Complete TypeIdentifier corresponding to the CompleteAnnotationType being registered.
+     * @param[out] type_ids TypeIdentifiers corresponding to the CompleteAnnotationType just registered and the
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp]: Done in eb4efe39b

File: include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp [hunk 17]
```diff
@@ -1719,15 +1755,16 @@ class TypeObjectUtils
...
+            TypeIdentifierPair& type_ids);
 
     /**
      * @brief Register structure TypeObject into TypeObjectRegistry.
      *        CompleteStructType is provided and the minimal TypeObject is constructed from the complete one.
      *
      * @param[in] struct_type CompleteStructType.
      * @param[in] type_name Name to be registered in the registry.
-     * @param[out] type_id Complete TypeIdentifier corresponding to the CompleteStructType being registered.
+     * @param[out] type_ids TypeIdentifiers corresponding to the CompleteStructType just registered and the
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp]: Done in eb4efe39b

File: include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp [hunk 18]
```diff
@@ -1738,15 +1775,16 @@ class TypeObjectUtils
...
+            TypeIdentifierPair& type_ids);
 
     /**
      * @brief Register union TypeObject into TypeObjectRegistry.
      *        CompleteUnionType is provided and the minimal TypeObject is constructed from the complete one.
      *
      * @param[in] union_type CompleteUnionType.
      * @param[in] type_name Name to be registered in the registry.
-     * @param[out] type_id Complete TypeIdentifier corresponding to the CompleteUnionType being registered.
+     * @param[out] type_ids TypeIdentifiers corresponding to the CompleteUnionType just registered and the
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp]: Done in eb4efe39b

File: include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp [hunk 19]
```diff
@@ -1757,14 +1795,16 @@ class TypeObjectUtils
...
-            TypeIdentifier& type_id);
+            TypeIdentifierPair& type_ids);
 
     /**
      * @brief Register bitset TypeObject into TypeObjectRegistry.
      *        CompleteBitsetType is provided and the minimal TypeObject is constructed from the complete one.
      *
      * @param[in] bitset_type CompleteBitsetType.
      * @param[in] type_name Name to be registered in the registry.
+     * @param[out] type_ids TypeIdentifiers corresponding to the CompleteBitsetType just registered and the
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp]: Done in eb4efe39b

File: include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp [hunk 20]
```diff
@@ -1833,6 +1874,8 @@ class TypeObjectUtils
      *
      * @param[in] enumerated_type CompleteEnumeratedType.
      * @param[in] type_name Name to be registered in the registry.
+     * @param[out] type_ids TypeIdentifiers corresponding to the CompleteEnumeratedType just registered and the
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp]: Done in eb4efe39b

File: include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp [hunk 21]
```diff
@@ -1842,14 +1885,17 @@ class TypeObjectUtils
...
+            const std::string& type_name,
+            TypeIdentifierPair& type_ids);
 
     /**
      * @brief Register bitmask TypeObject into TypeObjectRegistry.
      *        CompleteBitmaskType is provided and the minimal TypeObject is constructed from the complete one.
      *
      * @param[in] bitmask_type CompleteBitmaskType.
      * @param[in] type_name Name to be registered in the registry.
+     * @param[out] type_ids TypeIdentifiers corresponding to the CompleteBitmaskType just registered and the
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp]: Done in eb4efe39b

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp [hunk 22]
```diff
@@ -94,7 +94,8 @@ class TypeObjectRegistry : public ITypeObjectRegistry
      *
      * @param[in] type_name Name of the type being registered.
      * @param[in] complete_type_object CompleteTypeObject related to the given type name.
-     * @param[out] type_id TypeIdentifier corresponding to the CompleteTypeObject just registered.
+     * @param[out] type_ids TypeIdentifiers corresponding to the CompleteTypeObject just registered and the
```
[the remark this answers is not in the document]
Comment (richiware) [reply src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp]: Done in eb4efe39b

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp [hunk 23]
```diff
@@ -104,7 +105,31 @@ class TypeObjectRegistry : public ITypeObjectRegistry
...
+     *        TypeObjects constructing the minimal from the complete TypeObject information.
+     *        TypeObject consistency is not checked in this method as the order of the dependencies received by the
+     *        TypeLookupService is not guaranteed.
+     *        The consistency is checked by the TypeLookupService after all dependencies are registered.
+     *
+     * @pre @ref TypeIdentifierPair::type_identifier1 discriminator must match TypeObject discriminator or be TK_NONE.
+     *      @ref TypeIdentifierPair::type_identifier1 consistency is only checked in Debug build mode.
+     *
+     * @param[in] type_object Related TypeObject being registered.
+     * @param[inout] type_ids Returns the registered @ref TypeIdentifier.
```
[the remark this answers is not in the document]
Comment (richiware) [reply src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp]: Done in eb4efe39b

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp [hunk 24]
```diff
@@ -114,7 +139,8 @@ class TypeObjectRegistry : public ITypeObjectRegistry
      * @pre type_name must not be empty.
      *
      * @param[in] type_name Name of the type being registered.
-     * @param[in] type_identifier TypeIdentier related to the given type name.
+     * @param[inout] type_identifier TypeIdentifier related to the given type name. It must be set in
```
[the remark this answers is not in the document]
Comment (richiware) [reply src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp]: Done in eb4efe39b

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp [hunk 25]
```diff
@@ -174,20 +200,22 @@ class TypeObjectRegistry : public ITypeObjectRegistry
...
-     * @pre type_name must not be empty.
+     * @pre type_ids must not be empty. At least @ref TypeIdentifierPair::type_identifier1 must be filled.
      *
-     * @param[in] type_name Type which type information is queried.
-     * @param[out] type_information Related TypeInformation for the given type name.
-     * @return ReturnCode_t RETCODE_OK if the type_name is found within the registry.
-     *                      RETCODE_NO_DATA if the given type_name is not found.
-     *                      RETCODE_BAD_PARAMETER if the given type name corresponds to a indirect hash TypeIdentifier.
-     *                      RETCODE_PRECONDITION_NOT_MET if the type_name is empty.
+     * @param[in] type_ids @ref TypeIdentifier which type information is queried.
```
[the remark this answers is not in the document]
Comment (richiware) [reply src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp]: Done in eb4efe39b

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp [hunk 26]
```diff
@@ -231,38 +259,16 @@ class TypeObjectRegistry : public ITypeObjectRegistry
...
-    ReturnCode_t register_type_object(
-            const TypeIdentifier& type_identifier,
-            const TypeObject& type_object);
-
     /**
      * @brief Register DynamicType TypeObject.
      *
      * @param[in] dynamic_type DynamicType to be registered.
-     * @param[out] type_id TypeIdentifier corresponding to the registered DynamicType TypeObject.
+     * @param[out] type_ids TypeIdentifiers corresponding to the registered DynamicType TypeObject.
```
[the remark this answers is not in the document]
Comment (richiware) [reply src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp]: Done in eb4efe39b

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp [hunk 27]
```diff
@@ -772,144 +778,122 @@ class TypeObjectRegistry : public ITypeObjectRegistry
      * @brief Register DynamicType TypeObject of an Alias type.
      *
      * @param[in] dynamic_type Alias DynamicType to be registered.
-     * @param[out] type_id TypeIdentifier corresponding to the Alias DynamicType TypeObject.
-     *                     TypeIdentifier is required to define dependencies within the parent TypeObject
-     *                     (if applicable).
+     * @param[out] type_ids TypeIdentifiers corresponding to the Alias DynamicType TypeObject.
```
[the remark this answers is not in the document]
Comment (richiware) [reply src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp]: Done in eb4efe39b

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp [hunk 28]
```diff
@@ -772,144 +778,122 @@ class TypeObjectRegistry : public ITypeObjectRegistry
...
+            TypeIdentifierPair& type_ids);
 
     /**
      * @brief Register DynamicType TypeObject of an Annotation type.
      *
      * @param[in] dynamic_type Annotation DynamicType to be registered.
-     * @param[out] type_id TypeIdentifier corresponding to the Annotation DynamicType TypeObject.
-     *                     TypeIdentifier is required to define dependencies within the parent TypeObject
-     *                     (if applicable).
+     * @param[out] type_ids TypeIdentifiers corresponding to the Alias DynamicType TypeObject.
```
[the remark this answers is not in the document]
Comment (richiware) [reply src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp]: Done in eb4efe39b

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp [hunk 29]
```diff
@@ -772,144 +778,122 @@ class TypeObjectRegistry : public ITypeObjectRegistry
...
+            TypeIdentifierPair& type_ids);
 
     /**
      * @brief Register DynamicType TypeObject of a Structure type.
      *
      * @param[in] dynamic_type Structure DynamicType to be registered.
-     * @param[out] type_id TypeIdentifier corresponding to the Structure DynamicType TypeObject.
-     *                     TypeIdentifier is required to define dependencies within the parent TypeObject
-     *                     (if applicable).
+     * @param[out] type_ids TypeIdentifiers corresponding to the Alias DynamicType TypeObject.
```
[the remark this answers is not in the document]
Comment (richiware) [reply src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp]: Done in eb4efe39b

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp [hunk 30]
```diff
@@ -772,144 +778,122 @@ class TypeObjectRegistry : public ITypeObjectRegistry
...
+            TypeIdentifierPair& type_ids);
 
     /**
      * @brief Register DynamicType TypeObject of a Union type.
      *
      * @param[in] dynamic_type Union DynamicType to be registered.
-     * @param[out] type_id TypeIdentifier corresponding to the Union DynamicType TypeObject.
-     *                     TypeIdentifier is required to define dependencies within the parent TypeObject
-     *                     (if applicable).
+     * @param[out] type_ids TypeIdentifiers corresponding to the Alias DynamicType TypeObject.
```
[the remark this answers is not in the document]
Comment (richiware) [reply src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp]: Done in eb4efe39b

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp [hunk 31]
```diff
@@ -772,144 +778,122 @@ class TypeObjectRegistry : public ITypeObjectRegistry
...
+            TypeIdentifierPair& type_ids);
 
     /**
      * @brief Register DynamicType TypeObject of a Bitset type.
      *
      * @param[in] dynamic_type Bitset DynamicType to be registered.
-     * @param[out] type_id TypeIdentifier corresponding to the Bitset DynamicType TypeObject.
-     *                     TypeIdentifier is required to define dependencies within the parent TypeObject
-     *                     (if applicable).
+     * @param[out] type_ids TypeIdentifiers corresponding to the Alias DynamicType TypeObject.
```
[the remark this answers is not in the document]
Comment (richiware) [reply src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp]: Done in eb4efe39b

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp [hunk 32]
```diff
@@ -772,144 +778,122 @@ class TypeObjectRegistry : public ITypeObjectRegistry
...
+            TypeIdentifierPair& type_ids);
 
     /**
      * @brief Register DynamicType TypeObject of a Sequence type.
      *
      * @param[in] dynamic_type Sequence DynamicType to be registered.
-     * @param[out] type_id TypeIdentifier corresponding to the Sequence DynamicType TypeObject.
-     *                     TypeIdentifier is required to define dependencies within the parent TypeObject
-     *                     (if applicable).
+     * @param[out] type_ids TypeIdentifiers corresponding to the Alias DynamicType TypeObject.
```
[the remark this answers is not in the document]
Comment (richiware) [reply src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp]: Done in eb4efe39b

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp [hunk 33]
```diff
@@ -772,144 +778,122 @@ class TypeObjectRegistry : public ITypeObjectRegistry
...
+            TypeIdentifierPair& type_ids);
 
     /**
      * @brief Register DynamicType TypeObject of a Array type.
      *
      * @param[in] dynamic_type Array DynamicType to be registered.
-     * @param[out] type_id TypeIdentifier corresponding to the Array DynamicType TypeObject.
-     *                     TypeIdentifier is required to define dependencies within the parent TypeObject
-     *                     (if applicable).
+     * @param[out] type_ids TypeIdentifiers corresponding to the Alias DynamicType TypeObject.
```
[the remark this answers is not in the document]
Comment (richiware) [reply src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp]: Done in eb4efe39b

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp [hunk 34]
```diff
@@ -772,144 +778,122 @@ class TypeObjectRegistry : public ITypeObjectRegistry
...
+            TypeIdentifierPair& type_ids);
 
     /**
      * @brief Register DynamicType TypeObject of a Map type.
      *
      * @param[in] dynamic_type Map DynamicType to be registered.
-     * @param[out] type_id TypeIdentifier corresponding to the Map DynamicType TypeObject.
-     *                     TypeIdentifier is required to define dependencies within the parent TypeObject
-     *                     (if applicable).
+     * @param[out] type_ids TypeIdentifiers corresponding to the Alias DynamicType TypeObject.
```
[the remark this answers is not in the document]
Comment (richiware) [reply src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp]: Done in eb4efe39b

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp [hunk 35]
```diff
@@ -772,144 +778,122 @@ class TypeObjectRegistry : public ITypeObjectRegistry
...
+            TypeIdentifierPair& type_ids);
 
     /**
      * @brief Register DynamicType TypeObject of a Enumeration type.
      *
      * @param[in] dynamic_type Enumeration DynamicType to be registered.
-     * @param[out] type_id TypeIdentifier corresponding to the Enumeration DynamicType TypeObject.
-     *                     TypeIdentifier is required to define dependencies within the parent TypeObject
-     *                     (if applicable).
+     * @param[out] type_ids TypeIdentifiers corresponding to the Alias DynamicType TypeObject.
```
[the remark this answers is not in the document]
Comment (richiware) [reply src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp]: Done in eb4efe39b

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp [hunk 36]
```diff
@@ -772,144 +778,122 @@ class TypeObjectRegistry : public ITypeObjectRegistry
...
+            TypeIdentifierPair& type_ids);
 
     /**
      * @brief Register DynamicType TypeObject of a Bitmask type.
      *
      * @param[in] dynamic_type Bitmask DynamicType to be registered.
-     * @param[out] type_id TypeIdentifier corresponding to the Bitmask DynamicType TypeObject.
-     *                     TypeIdentifier is required to define dependencies within the parent TypeObject
-     *                     (if applicable).
+     * @param[out] type_ids TypeIdentifiers corresponding to the Alias DynamicType TypeObject.
```
[the remark this answers is not in the document]
Comment (richiware) [reply src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp]: Done in eb4efe39b

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp [hunk 37]
```diff
@@ -772,144 +778,122 @@ class TypeObjectRegistry : public ITypeObjectRegistry
...
+            TypeIdentifierPair& type_ids);
 
     /**
      * @brief Register DynamicType indirect-hash TypeIdentifier of a Sequence type.
      *
      * @param[in] dynamic_type Sequence DynamicType to be registered.
-     * @param[out] type_id Complete indirect hash TypeIdentifier corresponding to the Sequence DynamicType.
-     *                     TypeIdentifier is required to define dependencies within the parent TypeObject
-     *                     (if applicable).
+     * @param[out] type_ids TypeIdentifiers corresponding to the Alias DynamicType TypeObject.
```
[the remark this answers is not in the document]
Comment (richiware) [reply src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp]: Done in eb4efe39b

**Review by adriancampo — APPROVED** [review]
LGTM

**Review by adriancampo — APPROVED** [review]
LGTM

**Comment by EduPonz** [comment]
@richiprosima please test_3 discovery-server

**Comment by EduPonz** [comment]
@richiprosima please test_3 discovery-server

**Pull request merged**